cmd/node_agent/na/onprem_test.go

```diff
@@ -61,3 +61,50 @@ func TestGetServiceIdentity(t *testing.T) {
 		}
 	}
 }
+
+func TestGetTLSCredentials(t *testing.T) {
+	testCases := map[string]struct {
+		config      *Config
+		expectedErr string
+	}{
+		"Good cert": {
+			config: &Config{
+				CertChainFile:  "testdata/cert-from-root-good.pem",
+				KeyFile:        "testdata/key-from-root-good.pem",
+				RootCACertFile: "testdata/cert-root-good.pem",
+			},
+			expectedErr: "",
+		},
+		"Loading failure": {
+			config: &Config{
+				CertChainFile:  "testdata/cert-from-root-goo.pem",
```

Comment:
nit: Can you name it as testdata/cert-from-root-not-exist.pem?

Reply:
Done